ouroboros-network/src/Ouroboros/Network/BlockFetch/ClientState.hs

```diff
@@ -212,22 +212,37 @@ initialPeerFetchInFlight =
 
 addHeadersInFlight :: HasHeader header
                    => (header -> SizeInBytes)
-                   -> FetchRequest header
-                   -> Int
-                   -- ^ the number of new requests; this might differ from the
-                   -- length of 'FetchRequest' by one, since when we write to
-                   -- @TFetchRequestVar@ we combine 'FetchRequest's using their
-                   -- semigroup instance, which might merge two chain fragments.
+                   -> Maybe (FetchRequest header) -- ^ Any existing request.
+                   -> FetchRequest header         -- ^ The new (merged) request.
+                   -> FetchRequest header         -- ^ The extra request.
```

Comment:
The whole diff looks good to me. My only nitpick is here; I tripped over the use of the word "extra". Minor, just FYI.

We have (maybe) `old`, `new`, and `extra`, where `new = old <> extra` (or something close to that). But the word _extra_ initially sounds to me like "something else" that is not already in `new`. So maybe "latest"? "added"? "latestAdded"? Or so on.

Reply:
Thanks. And that's the relationship, and is a nice concise way to put it. I'll add that.